src/flb_engine_dispatch.c

```diff
@@ -183,10 +191,20 @@ static int tasks_start(struct flb_input_instance *in,
             hits++;
 
             /*
-             * We have the Task and the Route, created a thread context for the
-             * data handling.
+             * If the plugin doesn't allow for multiplexing but high throughput
```

Comment:
nit: I think this isn't the best place for an unclear remark about why this was added. I'd remove this comment, esp the "but high throughput is needed" bit

Reply:
IMO Either fully explain the option in a comment that is in a relevant and meaningful part of the code base or just don't have a comment

Reply:
Following the convention set forth here:
https://github.com/fluent/fluent-bit/blob/356a11b63ee4cae5d816aecac8814276cf57ea11/src/flb_engine_dispatch.c#L181-L189

I can remove this comment if you would like

Reply:
Move to the Macro
FLB_NO_MULTIPLEX only runs one task per flush cycle.
FLB_SYNCHRONOUS has no restriction on the amount of tasks ....